lib/std/Target.zig

```diff
@@ -2863,6 +3022,70 @@ pub fn cTypePreferredAlignment(target: Target, c_type: CType) u16 {
     );
 }
 
+pub fn defaultCCallingConvention(target: Target) ?std.builtin.CallingConvention {
+    return switch (target.cpu.arch) {
+        .x86_64 => switch (target.os.tag) {
+            .windows, .uefi => .{ .x86_64_win = .{} },
+            else => .{ .x86_64_sysv = .{} },
+        },
+        .x86 => switch (target.os.tag) {
+            .windows, .uefi => .{ .x86_win = .{} },
+            else => .{ .x86_sysv = .{} },
+        },
+        .aarch64, .aarch64_be => if (target.os.tag.isDarwin()) cc: {
+            break :cc .{ .aarch64_aapcs_darwin = .{} };
+        } else switch (target.os.tag) {
+            .windows => .{ .aarch64_aapcs_win = .{} },
+            else => .{ .aarch64_aapcs = .{} },
+        },
+        .arm, .armeb, .thumb, .thumbeb => .{ .arm_aapcs = .{} },
+        .mips64, .mips64el => switch (target.abi) {
+            .gnuabin32 => .{ .mips64_n32 = .{} },
+            else => .{ .mips64_n64 = .{} },
+        },
+        .mips, .mipsel => .{ .mips_o32 = .{} },
+        .riscv64 => .{ .riscv64_lp64 = .{} },
+        .riscv32 => .{ .riscv32_ilp32 = .{} },
+        .sparc64 => .{ .sparc64_sysv = .{} },
+        .sparc => .{ .sparc_sysv = .{} },
+        .powerpc64 => if (target.isMusl())
```

Comment:
This would normally be correct, but for now, it needs to always be ELFv2: https://github.com/ziglang/zig/blob/7b8fc18c666ba6952ad1571fbed4cc48e81f647d/src/target.zig#L421-L428

Reply:
Wow, that's stupid! I love LLVM so much!

If we're having this special case for now, probably we should just silently convert the callconv in the compiler itself when using the LLVM backend. This function in particular *must* live in `std.Target` since it's used to define `std.builtin.CallingConvention.c`, and it should not be tied to an implementation nor depend on the backend in use. This function should give the correct answer, and any workarounds for incorrect/incomplete implementations should live in the compiler.

Reply:
Ok, that works too.

In that case, can we rename it to `cCallingConvention()` or something similar that follows the established naming style for the other functions? `cTypeBitSize()`, `cTypeAlignment()`, etc.

Reply:
Yep, that's fair.